### PR TITLE
feat: payment box component 

### PR DESCRIPTION
**Pull request by alessandrogelmi**
## Short description
Implemented payment Box component

## List of changes proposed in this pull request
- Payment box in notificationDetail

## How to test
Describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

### CODE REVIEW

File: packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <Fragment>
+      <Typography variant="h6">
+        {getLocalizedOrDefaultLabel('notifications', 'detail.payment.title', 'Pagamento')}
+      </Typography>
+      <Typography variant="body2" sx={{ mt: 2 }}>
+        {getLocalizedOrDefaultLabel(
+          'notifications',
+          'detail.payment.subtitle',
+          'In questa notifica ci sono più avvisi di pagamento: seleziona quello che vuoi pagare. Alcuni avvisi includono i costi di notifica.'
+        )}{' '}
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx:24]: replace {' '} whit it's html analogus

File: packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        <Link href={void 0} target="_blank" fontWeight="bold" sx={{ cursor: 'pointer' }}>
+          {getLocalizedOrDefaultLabel('notifications', 'detail.payment.how', 'Come mai?')}
+        </Link>
+      </Typography>
+
+      <Button fullWidth variant="contained" sx={{ mt: 2 }}>
+        Paga
```
Comment (AndreaCimini90) [inline packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx:31]: Localize it

File: packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx [hunk 3]
```diff
@@ -93,7 +101,6 @@ describe('NotificationDetail Page', () => {
     expect(result.container).toHaveTextContent(/Table/i);
     expect(result.container).toHaveTextContent('detail.acts');
     expect(result.container).toHaveTextContent(/Timeline/i);
-    expect(result.container).toHaveTextContent(/Payment/i);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx:96]: this expect should be still valid

File: packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx [hunk 4]
```diff
@@ -105,7 +112,9 @@ describe('NotificationDetail Page', () => {
   });
 
   test('renders NotificationDetail page without payment box if noticeCode is empty', async () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx:114]: change the text and the test with new logic
`{ recipients: [{ payment: { noticeCode: '' } }] }`
is not valid
`{ recipients: [{ payments: [] / null }] }`
is a valid case

File: packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx [hunk 5]
```diff
@@ -4,7 +4,7 @@ import {
   apiOutcomeTestHelper,
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx:4]: use the mock in mocks folder

File: packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx [hunk 6]
```diff
@@ -124,7 +133,9 @@ describe('NotificationDetail Page', () => {
   });
 
   test('renders NotificationDetail page without payment box if creditorTaxId is empty', async () => {
-    result = await renderComponent(overrideNotificationMock({recipients: [{payment: { creditorTaxId: '' }}]}));
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx:127]: as before, change the test. the previous test can be payments null and this one can be payments empty

File: packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx [hunk 7]
```diff
@@ -143,7 +154,9 @@ describe('NotificationDetail Page', () => {
   });
 
   test('renders NotificationDetail page without payment box if noticeCode and creditorTaxId are both empty', async () => {
-    result = await renderComponent(overrideNotificationMock({recipients: [{payment: { creditorTaxId: '', noticeCode: '' }}]}));
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx:146]: as before, we don't have payment any more

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx (hunk 1 above)
Comment (alessandrogelmi) [reply packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx:24]: `&nbsp;` this one?

Comment (AndreaCimini90) [reply packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx:24]: yes

File: packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import { getLocalizedOrDefaultLabel } from '../../services/localization.service';
+import { PaymentHistory } from '../../types';
+
+type Props = {
+  loading: boolean;
+  payments: Array<PaymentHistory>;
+};
+
+const NotificationPaymentRecipient: React.FC<Props> = ({ loading }) => {
+  console.log(loading);
```
[the remark this answers is not in the document]
Comment (alessandrogelmi) [reply packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx:13]: Skeleton will be in children component

**Review by AndreaCimini90 — APPROVED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**